gqcp/include/FockSpace/WaveFunction/WaveFunctionSelection.hpp

```diff
@@ -0,0 +1,76 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include "Basis/TransformationMatrix.hpp"
+#include "FockSpace/SelectedFockSpace.hpp"
+#include "FockSpace/WaveFunction/WaveFunction.hpp"
+#include "Mathematical/Representation/Matrix.hpp"
+
+
+
+namespace GQCP {
+
+
+/**
+ *  A class that represents a selection of a wave function
+ */
+class WaveFunctionSelection {
+private:
+    std::vector<Configuration> configurations;
+    VectorX<double> coefficients;  // the coefficients for the ONV configurations
+
+public:
+    // CONSTRUCTORS
+    WaveFunctionSelection() = default;
+
+    /**
+     *  Construct a selected wave function
+     *
+     *  @param fock_space           the selected Fock space in which the wave function 'lives'
+     *  @param coefficients         the expansion coefficients
+     */
+    WaveFunctionSelection(const SelectedFockSpace& fock_space, const VectorX<double>& coefficients);
+
+
+    /**
+     *  Construct a selected wave function by selecting the configurations with the highest contribution from a different wave function
+     *
+     *  @param wave_function                           the wave_function to be selected from
+     *  @param number_of_largest_contributions         the amount of selections made
+     */
+    WaveFunctionSelection(const WaveFunction& wave_function, size_t number_of_largest_contributions);
```

Comment:
I find it very awkward as an API to have to invoke the constructor.

It's really a lot more natural if the API would be

```cpp
WaveFunctionSelection WaveFunction::select(const size_t num) const;
```

where `num` is the number of configurations that are chosen based on their associated coefficient in the wave function expansion.

Reply:
I agree, this would make sense, this would be more of a struct then